FushaarProvider/src/main/kotlin/com/fushaar/FushaarProvider.kt

```diff
@@ -0,0 +1,135 @@
+import com.lagradost.cloudstream3.utils.getQualityFromName
+import org.jsoup.nodes.Element
+
+class FushaarProvider : MainAPI() {
+    override var lang = "ar"
+    override var mainUrl = "https://www.fushaar.com"
+    override var name = "Fushaar"
+    override val usesWebView = false
+    override val hasMainPage = true
+    override val supportedTypes = setOf(TvType.Movie)
+
+
+    private fun String.getIntFromText(): Int? {
+        return Regex("""\d+""").find(this)?.groupValues?.firstOrNull()?.toIntOrNull()
+    }
+
+    private fun Element.toSearchResponse(): SearchResponse? {
+        val url = select("article.poster")
+        val posterUrl = select("img").attr("data-lazy-src")
+        val year = select("ul.labels li.year").text()?.getIntFromText()
+        var quality = select("div").first()?.attr("class")?.replace("hdd","hd")?.replace("caam","cam")
+        val title = select("div.info h3").text()+"\n"+select("div.info h4").text()
+
+        return MovieSearchResponse(
+            title,
+            url.select("a").attr("href"),
+            this@FushaarProvider.name,
+            TvType.Movie,
+            posterUrl.also{print("posterurl :"+it)},
+            year,
+            null,
+            quality = getQualityFromString(quality),
+        )
+    }
+
+    override suspend fun getMainPage(page: Int, request : MainPageRequest): HomePageResponse {
+        // Title, Url
+        val moviesUrl = listOf(
+            "Movies" to "$mainUrl/page/" + (0..25).random(),
+            "Herror" to "$mainUrl/gerne/herror",
+            "Thriller" to "$mainUrl/gerne/thriller",
+            "Action" to "$mainUrl/gerne/action",
+            "Animation" to "$mainUrl/gerne/animation",
+            "Comedy" to "$mainUrl/gerne/comedy",
+            "Sci-fi" to "$mainUrl/gerne/sci-fi",
+            "Crime" to "$mainUrl/gerne/crime",
+            "Drama" to "$mainUrl/gerne/drama",
+            "Adventure" to "$mainUrl/gerne/adventure",
+            "Biography" to "$mainUrl/gerne/biography",
+            "Music" to "$mainUrl/gerne/music",
+            "Sport" to "$mainUrl/gerne/sport",
+            "Documentary" to "$mainUrl/gerne/documentary",
+            "History" to "$mainUrl/gerne/history",
+            "Family" to "$mainUrl/gerne/family",
+            "Romance" to "$mainUrl/gerne/romance",
+            "Mystery" to "$mainUrl/gerne/mystery"
+        )
+        val pages = moviesUrl.apmap {
+            val doc = app.get(it.second).document
+            val list = doc.select("article.poster").mapNotNull { element ->
+                element.toSearchResponse()
+            }
+            HomePageList(it.first, list)
+        }.sortedBy { it.name }
+        return HomePageResponse(pages)
+    }
+
+    override suspend fun search(query: String): List<SearchResponse> {
+        val q = query.replace(" ", "%20")
+        val result = arrayListOf<SearchResponse>()
+        listOf(
+            "$mainUrl/?s=$q",
+        ).apmap { url ->

```

Comment:
I think no need for apmap here? just do the request alone since its just one url
```kotlin
	return app.get("$mainUrl/?s=$q").document.select("article.poster").mapNotNull {
                it.toSearchResponse()?.let { it1 -> result.add(it1) }
            }
        }
```

Reply:
remove ?.let { it1 -> result.add(it1) } and result